### PR TITLE
API-20647 cleanup prep OIDC dep removal

### PR DESCRIPTION
**Pull request by crolarlibertyva**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- Preps for the removal of internal token logic
- Removes unused logic from the ValidatedToken class

## Related issue(s)

- https://jira.devops.va.gov/browse/API-20647

## Testing done

- Basic claims api test
- Claims rspec modified

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
* Claims API

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/claims_api/app/models/claims_api/validated_token.rb [hunk 1]
```diff
@@ -26,37 +23,6 @@ def initialize(token_validation_url, token_string, audience)
...
-      key
-    rescue JWT::DecodeError => e
-      raise error_klass("Unable to determine public key: #{e.message}")
-    end
-
-    def error_klass(error_detail_string)
-      # If there is an error validating the token, raise it here
-      ::Common::Exceptions::TokenValidationError.new(detail: error_detail_string)
-    end
-
```
Comment (FonzMP) [inline modules/claims_api/app/models/claims_api/validated_token.rb:59]: @crolarlibertyva - I believe this was added back as our consumers reported a change in the response for v1. @rockwellwindsor-va can probably provide more details but we were seeing a 401 unauthorized, but it was reported that v1 (before the changes we introduced a month or so ago) sent back a "Signature has expired." message. If we're looking at this as the new intent, it's fine but we would need to relay that to consumers before making this change.

Comment (FonzMP) [reply modules/claims_api/app/models/claims_api/validated_token.rb:59]: From our consumer:

```
Here is a sample error response from Aug 30:

{
"errors": [

{ "title": "Not authorized", "detail": "Signature has expired", "code": "401", "status": "401" }
]
}

Here is a sample error response from Aug 31:

{
"errors": [

{ "title": "Not authorized", "detail": "Not authorized", "code": "401", "status": "401" }
]
}
```

Comment (crolarlibertyva) [reply modules/claims_api/app/models/claims_api/validated_token.rb:59]: Let me look into processing the error response from the validation call to pass along the error detail to indicate 'signature  expired', and other items like "invalid audience".

Comment (crolarlibertyva) [reply modules/claims_api/app/models/claims_api/validated_token.rb]: @FonzMP : Updated and confirmed the response to be 
```
{
  "errors": [
    {
      "title": "Not authorized",
      "detail": "Signature has expired",
      "code": "401",
      "status": "401"
    }
  ]
}
```

File: modules/claims_api/app/models/claims_api/validated_token.rb [hunk 2]
```diff
@@ -64,5 +30,17 @@ def client_credentials_token?
...
     end
+
+    def call_token_validation_service(token_validation_url, payload)
+      RestClient.post(token_validation_url,
+                      payload,
+                      { Authorization: "Bearer #{@token_string}",
+                        apiKey: Settings.claims_api.token_validation.api_key })
+    rescue => e
+      error = JSON.parse(e.response)
+      err_detail = error['errors'][0]['detail']
```
Comment (dbrown-va) [inline modules/claims_api/app/models/claims_api/validated_token.rb:41]: This seems like it has null pointer potential, but I am not a rails expert.

File: modules/claims_api/app/models/claims_api/validated_token.rb [hunk 3]
```diff
@@ -64,5 +30,17 @@ def client_credentials_token?
...
+      err_detail = 'Signature has expired' if err_detail.include?('expired')
```
Comment (dbrown-va) [inline modules/claims_api/app/models/claims_api/validated_token.rb:42]: This string seems like it _could_ be a candidate for an enum?

File: modules/claims_api/app/models/claims_api/validated_token.rb (hunk 2 above)
Comment (crolarlibertyva) [reply modules/claims_api/app/models/claims_api/validated_token.rb:41]: Will address. Better to be safe.

**Review by dbrown-va — APPROVED** [review]
Agreed with this change as the oidc/key_service is going to be extracted unless other usages are identified.

It would be great if we could reference errors in a more constant / enum driven form, but that can be addressed as the Claims team prefers as code owners.

File: modules/claims_api/app/models/claims_api/validated_token.rb (hunk 1 above)
Comment (crolarlibertyva) [reply modules/claims_api/app/models/claims_api/validated_token.rb]: So the original response is now intact in the case of an expired token.

File: modules/claims_api/app/models/claims_api/validated_token.rb (hunk 2 above)
Comment (crolarlibertyva) [reply modules/claims_api/app/models/claims_api/validated_token.rb]: Done.

**Review by stiehlrod — APPROVED** [review]
Tested the following endpoints:
- {{uri}}/services/claims/v2/veterans/1012845630V900607/526
- {{uri}}/services/claims/v2/veterans/1012845630V900607/claims/600437016
- {{uri}}/services/claims/v2/veterans/{{veteran_id}}/claims/600016536/5103
- https://staging-api.va.gov/services/veteran/v0/representatives/find_rep?first_name={{veteran_first_name}}&last_name={{veteran_last_name}}
- {{uri}}/services/claims/v2/veterans/{{veteran_id}}/intent-to-file/survivor
- {{uri}}/services/claims/v1/forms/526
- {{uri}}/services/claims/v1/claims/600098193

**Review by FonzMP — APPROVED** [review]
:shipit: thanks @crolarlibertyva

**Pull request merged**